graph/src/components/subgraph/registry.rs

```diff
@@ -15,33 +15,57 @@ impl<T> Clone for SubgraphRegistry<T> {
     }
 }
 
-impl<T> SubgraphRegistry<T>
-where
-    T: Clone + Send,
-{
+impl<T> SubgraphRegistry<T> {
     pub fn new() -> Self {
         SubgraphRegistry {
             names_to_ids: Default::default(),
             ids_to_values: Default::default(),
         }
     }
 
-    pub fn resolve(&self, id_or_name: &String) -> Option<T> {
+    pub fn resolve_by<U, F: FnOnce(&T) -> U>(&self, id_or_name: &String, by: F) -> Option<U> {
```

Comment:
Does this extra `map(by)` call at the end really warrant a dedicated method? It would seem trivial to just add it to via `subgraphs.resolve(&subgraph).map(|s| s.schema.clone())`.

Reply:
I see that the regular `resolve` clones at the end. How about we make that return `Option<&T>` and then we can clone/map where we need to outside the method?

Reply:
_If_ we decide to keep this, I'd prefer `resolve_map` (like `filter_map`).

Reply:
I tried making it return a reference but couldn't get it to work because the reference is limited by the lifetime of the lock, even if I also returned the lock it still doesn't work. The rename sounds good.